src/pages/send/confirm/confirm.ts

```diff
@@ -727,7 +777,13 @@ export class ConfirmPage extends WalletTabsChild {
       }
 
       this.walletProvider
-        .createTx(wallet, txp)
+        .getAddress(this.wallet, false)
```

Comment:
To avoid the cascade of "then" and be more consistent with the Copay code, you could write this function as:
```this.walletProvider
        .getAddress(this.wallet, false)
        .then(address => {
          txp.from = address;
          this.walletProvider.createTx(wallet, txp).then(ctxp => {
            return resolve(ctxp);
          }).catch(err => {
            return reject(err);
          });
        }).catch(err => {
          return reject(err);
        });```

or even remove `return new Promise((resolve, reject) => {`
and return a promise function directly, and use Promise.reject() / Promise.resolve()

Reply:
Code changed to the comment above!